src/components/sidebar/types.ts

```diff
@@ -1,12 +1,105 @@
 import { SidebarBackToLinkProps } from './components/sidebar-back-to-link'
 
+/**
+ * Interfaces for each nav item in their raw form, before any enrichment or
+ * filtering via the filter input
+ */
```

Comment:
For consideration: I've tended to use a `Raw` prefix when typing data that I'm getting from a source file if I'm planning to enhance / transform it further. I see the word `raw` in this comment, so that made me think it could make sense here!

ref:
* https://github.com/hashicorp/mktg-content-workflows/blob/a75088422dab6324e6c36adbd168162499ed700b/workflows/extract-content/utils/extract-documents.ts#L5
* https://github.com/hashicorp/mktg-content-workflows/blob/7325e75bff02671d935628cdb69baf64a66157c5/workflows/extract-content/utils/extract-nav-data.ts#L13

Reply:
Love this idea! I'll update. 😊

Reply:
ooh i like this `raw` prefix idea 💯 thanks for sharing it! ❤️

Reply:
I'd rather not rename, as I feel like the names defined here flow more intuitively. I'll add a comment that mentions what the `Raw` types are based off of!

Reply:
Added more comments in https://github.com/hashicorp/dev-portal/pull/326/commits/1d61cdaa31b5a772a9eea05723d107fbd6456448!